### ISSUE TITLE
Position gets wrongly re-calculated after scroll and onLayout (using SafeAreaView)

### ISSUE THREAD

**Issue by harunsmrkovic** [react-native-community/react-native-tab-view#625]
### Current behaviour

Say we have two top-level screens, Screen A with `SafeAreaView > TabView` inside, and Screen B with whatever else. If you go to Screen A, then swipe to go to another tab, then go to Screen B, and then back to Screen A, `position` will be wrong. It will either be `-1` or will be pointing to another element being selected.

### Expected behaviour

`position` should always reflect currently selected index.

### Code sample

```
import { createSwitchNavigator } from 'react-navigation'

const MainNavigator = createSwitchNavigator({
  screenA: { screen: ScreenA },
  screenB: { screen: ScreenB }
})

const ScreenA = () => (
  <SafeAreaView>
    <TabView {...whateverProps} />
  </SafeAreaView>
)

const ScreenB = () => <Text>Whatever</Text>
```

### What have you tried

I have found the issue to be in the `handleLayout` function, in TabView. If values of offset/panX get updated via `handleScroll`, and `handleLayout` gets called afterwards, it messes up the rendering...
`handleLayout` is being called on focusing back to `TabView` because it is wrapped into `SafeAreaView`.

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | ios
| react-native          | 0.55.3
| react-native-tab-view | ^1.2.0
| node                  | v8.10.0
| npm or yarn           | yarn

**Comment by harunsmrkovic** [comment]
@satya164 hey, can you please check it out? I can assist if something's missing

**Comment by satya164** [comment]
Please provide a runnable example.

**Comment by harunsmrkovic** [comment]
@satya164 updated the original description

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
